### ISSUE TITLE
Issue with redirects in s3.aws.com urls

### ISSUE THREAD

**Issue by heglund** [waderobson/s3-auth#1]
There seems to be an issue with redirections depending on the URL of the S3 bucket.

For example, if a bucket has a name blah.s3.amazonaws.com, S3 may return a HTTP 307 when requested. This can happen if your AWS region is anything but us-east.

If you are using the ap-southeast-2 region, and request blah.s3.amazonaws.com, you are redirected to blah.s3-ap-southeast-2.amazonaws.com

As a result, you end up with this error upon running managedsoftwareupdate:

```
Retrieving list of software for this machine...
ERROR: Could not retrieve manifest YOUR_MANIFEST from the server: HTTP result 307: temporarily redirected
ERROR: Could not retrieve managed install primary manifest.
Finishing.
```

As a workaround, you can modify the following function to look for a different string. Instead of `if 's3.amazonaws.com' in options['url']:` you could substitute region specific information such as `if 's3-ap-southeast-2.amazonaws.com' in options['url']:`

```
def process_request_options(options):
    """Make changes to options dict and return it.
       This is the fuction that munki calls."""
    if 's3.amazonaws.com' in options['url']:
        headers = s3_auth_headers(options['url'])
        options['additional_headers'].update(headers)
    return options
```

So, two options:
1. Add region info to that search string from the region data already in /Library/Preferences/com.github.waderobson.s3-auth.plist
2. Add support for URL redirection adding headers to anything directed from *.s3.amazonaws.com

**Comment by waderobson** [comment]
Interesting. I will try moving some stuff to that region so I can check this out. I haven't' run into this  using the us-west-2 region so us-east is not the only one that is working out of the box. 

Have you tried changing the follow redirects preference for munki. Just curious.
https://github.com/munki/munki/wiki/Preferences#followhttpredirects

**Comment by heglund** [comment]
Hey @waderobson,

Yeah, I had tried playing around with the followhttpredirects option after I posted this. It still seemed problematic, but in looking at it this morning, I think I've found the root cause. It's to do with the S3 DNS propagation.

http://docs.aws.amazon.com/AmazonS3/latest/dev/Redirects.html

The buckets that I was testing on were about 5 hours old, but were still being redirected to a new location. The AWS doc above suggests that any programs making S3 requests should be able to follow an HTTP redirect request (even though it lists status 302 not 307).

Testing this morning, all works with the default code! Unlikely to be a major problem for anyone else as long as their bucket has been around for a while, but I might have look on Monday to see if I can make a more predictable test, or suggest something for handling the redirect more gracefully.

Thanks for your work on this - proving to be an incredibly useful addition to Munki.
